### DISCUSSION TITLE
doublets removeal

### DISCUSSION THREAD

**Discussion by ZengMuo** [stuart-lab/signac#1736] [Q&A]
Hi, Tim. How do you perform doublet removal for scATAC?

**Comment by Mialn20** [comment]
Not an expert, but I have used this package:
https://github.com/plger/scDblFinder
https://plger.github.io/scDblFinder/articles/scATAC.html


# Seurat Object
Samples <- c(SampleObject)

# Making single cell experiment object and running scDblFinder
scEX_SampleObject <- as.SingleCellExperiment(SampleObject)
scEX_SampleObject <- scDblFinder(scEX_SampleObject,aggregateFeatures=TRUE,nfeatures=25, processing="normFeatures") 


class <- scEX_SampleObject$scDblFinder.class # Classification 
SampleObject$Class <- class
score <- scEX_SampleObject$scDblFinder.score #The final doublet score
SampleObject$Score <- score
weight <- scEX_SampleObject$scDblFinder.weighted
SampleObject$Weight <- weight

plot_grid(
  DimPlot(SampleObject, reduction = "umap", group.by = "Class") + ggtitle(paste0("SampleObject - ", ncol(subset(SampleObject, subset = Class == "singlet")), " cells remains - doublets removed: ",sum(SampleObject$Class=="doublet"))),
  ,ncol=2)
ggsave(paste0("File path here.pdf"),
       height=15, width=15)

# Subsetting the singlets for the seurat objects
SampleObject<- subset(SampleObject, subset = Class == "singlet")
